src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -116,97 +136,56 @@ public Command parseCommand(String userInput) throws ParseException {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.ADD;
-            }
-        }
-        for (String word : AddLifeInsuranceCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.ADDLI;
-            }
-        }
-        for (String word : ClearCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.CLEAR;
-            }
-        }
-        for (String word : DeleteCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.DEL;
-            }
-        }
-        for (String word : EditCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.EDIT;
-            }
-        }
-        for (String word : ExitCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.EXIT;
-            }
-        }
-        for (String word : FindCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.FIND;
-            }
+        if (commandWordMap.isEmpty()) {
+            setUpCommandWordMap();
         }
-        for (String word : PartialFindCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.PFIND;
-            }
-        }
-        for (String word : HelpCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.HELP;
-            }
-        }
-        for (String word : HistoryCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.HISTORY;
-            }
-        }
-        for (String word : ListCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.LIST;
-            }
-        }
-        for (String word : RedoCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.REDO;
-            }
-        }
-        for (String word : SelectCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.SELECT;
-            }
-        }
-        for (String word : UndoCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.UNDO;
-            }
-        }
-
-        for (String word : WhyCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.WHY;
-            }
-        }
-        for (String word : PrintCommand.COMMAND_WORDS) {
-            if (commandWord.contentEquals(word)) {
-                return CommandType.PRINT;
-            }
-        }
-
-        return CommandType.NONE;
+        return commandWordMap.getOrDefault(commandWord, CommandType.NONE);
```

Comment:
IntelliJ is really intelligent on this lol, i tried to use the `condition ? if : else` method but it told me why not use getOrDefault. Mindblown.